react-frontend/src/core-components/Button/ButtonComponent.js

```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { Button } from "react-bootstrap";
+import PropTypes from "prop-types";
+
+const ButtonComponent = ({ className, type, onClick, size, text, icon }) => (
+  <Button className={className} type={type} onClick={onClick} size={size}>
+    {icon}
+    {text}
```

Comment:
what if I have a requirement where I am using icon in a button with no text? image a download, upload button. I don't want text required prop error